File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/actions/AbstractGameActionWithAckMessage.scala





















// Generated on 05/03/2013 10:34:32
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait AbstractGameActionWithAckMessage extends AbstractGameActionMessage {
  override val messageId = 1001

  protected trait Message extends AbstractGameActionMessage.Message with Serializable {
    override val messageId = AbstractGameActionWithAckMessage.messageId

    val waitAckId: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(waitAckId)
    }

  }

  protected def create(actionId: Short, sourceId: Int, waitAckId: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val actionId = buf.readShort()
    val sourceId = buf.readInt()
    val waitAckId = buf.readShort()

    create(actionId, sourceId, waitAckId)
  }

}

object AbstractGameActionWithAckMessage extends AbstractGameActionWithAckMessage {
  case class MessageImpl(actionId: Short, sourceId: Int, waitAckId: Short) extends Message
  type Target = MessageImpl

  override def create(actionId: Short, sourceId: Int, waitAckId: Short) = MessageImpl(actionId, sourceId, waitAckId)

  def apply(actionId: Short, sourceId: Int, waitAckId: Short) = create(actionId, sourceId, waitAckId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/AbstractGameActionWithAckMessage.java


// Generated on 05/08/2013 19:37:38
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AbstractGameActionWithAckMessage extends AbstractGameActionMessage {
    public static final int MESSAGE_ID = 1001;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short waitAckId;
    
    public AbstractGameActionWithAckMessage() { }
    
    public AbstractGameActionWithAckMessage(short actionId, int sourceId, short waitAckId) {
        super(actionId, sourceId);
        this.waitAckId = waitAckId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(waitAckId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        waitAckId = buf.readShort();
    }
    
}
